airbyte-workers/src/main/java/io/airbyte/workers/DefaultDiscoverCatalogWorker.java

```diff
@@ -87,7 +90,21 @@ public DefaultDiscoverCatalogWorker(final IntegrationLauncher integrationLaunche
       int exitCode = process.exitValue();
       if (exitCode == 0) {
         if (catalog.isEmpty()) {
-          LOGGER.error("integration failed to output a catalog struct.");
+          LOGGER.error("Integration failed to output a catalog struct.");
+          return new OutputAndStatus<>(JobStatus.FAILED);
+        }
+
+        List<String> invalidStreamNames = CatalogHelpers.getInvalidStreamNames(catalog.get());
+
+        if (!invalidStreamNames.isEmpty()) {
+          invalidStreamNames.forEach(streamName -> LOGGER.error("Cannot sync invalid stream name: " + streamName));
+          return new OutputAndStatus<>(JobStatus.FAILED);
```

Comment:
can we also return a message describing why discovery failed so we can show this on the UI?

Reply:
`OutputAndStatus` doesn't allow for error messages unfortunately. Adding messages outside of logs seems a bit out of scope for this PR. https://github.com/airbytehq/airbyte/issues/1011

Reply:
I understand that, but I'm just worried that in its current state, there is no way for the user to know what's wrong. It'll just look like airbyte is broken if e.g: they sync Gsheets with a space in the name, and all they'll see on the UI is "an error happened"